src/main.rs

```diff
@@ -166,13 +166,19 @@ fn compile(args: Cli) {
     // }
 
     // Phase 6: Codegen
-    if args.opt_level == '0' && args.backend != Some(Backend::Llvm) {
-        cranelift_backend::run(filename, hir, &args);
-    } else if cfg!(feature = "llvm") {
-        #[cfg(feature = "llvm")]
-        llvm::run(filename, hir, &args);
-    } else {
-        eprintln!("The llvm backend is required for non-debug builds. Recompile ante with --features 'llvm' to enable optimized builds.");
+    let backend =
+        args.backend.unwrap_or_else(|| if args.opt_level == '0' { Backend::Cranelift } else { Backend::Llvm });
```

Comment:
This reads somewhat worse to me to be honest, I think we should revert it. Perhaps it can be revisited if there is ever a third backend or the logic becomes more complex